### PR TITLE
hidden bcast_params call in dist train

### LINKED ISSUE #11572: Hidden parallel_executor.bcast_params() when distributed train

### PR DESCRIPTION
**Pull request by Yancey0623**
Fixed #11572

### CODE REVIEW

File: python/paddle/fluid/parallel_executor.py [hunk 1]
```diff
@@ -131,6 +132,11 @@ def __init__(self,
         main = main_program
         main = main if main else framework.default_main_program()
         scope = executor.global_scope()
+        # FIXME(Yancey1989): it's a temporary approach to determinate the distribute
+        # train program, call self.bcast_param() at the end of each mini-batch.
+        self.is_dist = True if "recv" in [
```
Comment (typhoonzero) [inline python/paddle/fluid/parallel_executor.py:134]: Can use num_trainers to test if it's distributed training

Comment (panyx0718) [reply python/paddle/fluid/parallel_executor.py:134]: The ```num_trainers``` looks a bit suspicious now. Is it only for nccl? I found some pserver training script working without setting num_trainers...

Comment (Yancey0623) [reply python/paddle/fluid/parallel_executor.py:134]: It's only a temporary fix, we can insert `bcast` after `recv` op in PrallelExecuotor.

Comment (Yancey0623) [reply python/paddle/fluid/parallel_executor.py:134]: And i think @panyx0718 is right, pserver update does not set num_trainers.

**Review by panyx0718 — DISMISSED** [review]
Added a issue to address this properly soon: https://github.com/PaddlePaddle/Paddle/issues/11593

**Review by panyx0718 — APPROVED** [review]

**Pull request merged**